.travis.yml

```diff
@@ -1,68 +1,67 @@
+---
 language: node_js
 node_js:
-- '4'
+  # we recommend testing addons with the same minimum supported node version as Ember CLI
+  # so that your addon works for all apps
+  - "8"
+
 sudo: false
+dist: trusty
+
+addons:
+  chrome: stable
 
 cache:
   yarn: true
-  directories:
-    - $HOME/.cache # includes bower's cache
 
 env:
-  matrix:
-    - EMBER_TRY_SCENARIO=default
-    - EMBER_TRY_SCENARIO=ember-2.0
-    - EMBER_TRY_SCENARIO=ember-lts-2.4
-    - EMBER_TRY_SCENARIO=ember-lts-2.8
-    - EMBER_TRY_SCENARIO=ember-lts-2.12
-    - EMBER_TRY_SCENARIO=ember-lts-2.16
-    - EMBER_TRY_SCENARIO=ember-release
-    - EMBER_TRY_SCENARIO=ember-beta
-    - EMBER_TRY_SCENARIO=ember-canary
   global:
-    - GITHUB_USER=tomster-strudel
-    - secure: A5DKT8O+wmwEY3ReKkBMFgy+r2EaUgxMd9FMR+MOGLWMzVdEm6QX0tQopWgqpoPLpymr6aUNMYrb3mRoYygBpUaobgrRRh2f+IGQ+i65+gqC+Y8ma8sArbAKVZ+QuV42ICPGTtk2ep9nQ81k1NO714iOaok4jBuTPFrIOEFx4NSWOuxAHSyjpb41YcoZ4420taXOT6PEtU528lAeLTZeI/gfn/TyDFS5Hf+CoFHxZgLZ+gtdJugu4+11KLnOPgCsAmL2AKagk3LQzfHgaASME4GqXDwz5iBloMWeP2RJJQxwUtB/XgPz//VLh1nOdfodhRy0pjeRzrgvoMYiyqKvG1UMCqjqz/WEYtdeR335NC3kPv9tCcMCeN6vvCaKoang9i+LGEFLzj+zN50ldLMfkF/YEu/PkmezYLWpQwZdlwGQpZjLabT0etS46rdSIDU3M/4N4Waj3o9G8ZXgx7fTOd9ypRPbvOBslUlzup+jMLfGoKLcdT7Cmd3qTJ7XAWFRTTGA/fM7Bd8SYY1FQBAEsHH+2HqF0RsdSAnNpXQtVdzs5TjS9jzLM2rG4S2p0Dly7fe0KgYBD/X21ZDq8c44RRKCFFNI4GYGDA4yS6C6Sy3DVGv2yTQpB2zjvHG37HEjPtlvLYvuEWrYPnYUDEqJ/HRMN3qlODfyr1/+5+4+k8E=
+    # See https://git.io/vdao3 for details.
+    - JOBS=1
+
+branches:
+  only:
+    - master
+    # npm version tags
+    - /^v\d+\.\d+\.\d+/
 
-matrix:
-  fast_finish: true
+jobs:
+  fail_fast: true
   allow_failures:
-    - env: EMBER_TRY_SCENARIO=ember-2.0 # TODO fix this
-    - env: EMBER_TRY_SCENARIO=ember-lts-2.12 # TODO fix this
-    - env: EMBER_TRY_SCENARIO=ember-lts-2.16 # TODO fix this
-    - env: EMBER_TRY_SCENARIO=ember-release # TODO fix this
-    - env: EMBER_TRY_SCENARIO=ember-beta # TODO fix this
     - env: EMBER_TRY_SCENARIO=ember-canary
 
+  include:
+    # runs linting and tests with current locked deps
+
+    - stage: "Tests"
+      name: "Tests"
+      install:
+        - yarn install --non-interactive
+      script:
+        - yarn lint:hbs
+        - yarn lint:js
+        - yarn test
+
+    - name: "Floating Dependencies"
+      script:
+        - yarn test
+
+    # we recommend new addons test the current and previous LTS
+    # as well as latest stable release (bonus points to beta/canary)
+    - stage: "Additional Tests"
+      env: EMBER_TRY_SCENARIO=ember-lts-3.4
+    - env: EMBER_TRY_SCENARIO=ember-lts-3.8
+    - env: EMBER_TRY_SCENARIO=ember-release
+    - env: EMBER_TRY_SCENARIO=ember-beta
+    - env: EMBER_TRY_SCENARIO=ember-canary
```

Comment:
We should add test with different ember-data versions

Reply:
Done, I think